android/src/main/cpp/graphics/shader/BaseShaderProgramOpenGl.cpp

```diff
@@ -114,4 +156,24 @@ void BaseShaderProgramOpenGl::preRender(const std::shared_ptr<::RenderingContext
             break;
         }
     }
-}
\ No newline at end of file
+
+    std::shared_ptr<OpenGlContext> openGlContext = std::dynamic_pointer_cast<OpenGlContext>(context);
+    if (openGlContext) {
+        if (program == GL_INVALID_INDEX) {
+            program = openGlContext->getProgram(getProgramName());
+        }
+
+        if (frameUniformsBufferBlockIdx == GL_INVALID_INDEX) {
+            frameUniformsBufferBlockIdx = glGetUniformBlockIndex(program, "FrameUniforms");
+            if (frameUniformsBufferBlockIdx == GL_INVALID_INDEX) {
+                LogError <<= "Uniform block FrameUniforms not found - update shader to use the frame-wide uniform values.";
+                frameUniformsBufferBlockIdx = -2;
+            } else {
+                glUniformBlockBinding(program, frameUniformsBufferBlockIdx, FRAME_UBO_BINDING_POINT);
```

Comment:
Here and for all other uniform blocks. We specify `layout(binding = ..)` in the shader. Isn't this re-binding here redundant?
FWIW, WebGL does not support `layout(binding = ..)` natively, so I'd be happy if we just leave this out, to keep the diff for webgl smaller. But the redundancy is confusing.